### ISSUE TITLE
[spec] Add skipped counter to spec report

### ISSUE THREAD

**Issue by straight-shoota** [crystal-lang/crystal#8403]
The statistics line in spec reports should get a `skipped`  count:

```
2 examples, 1 failures, 0 errors, 0 pending, 3 skipped
```

Skipped is different from pending because pending are specs that are not even implemented, but skipped specs are just skipped for this run.

Examples are skipped when:
* `--fail-fast` option stops execution at the first failure
* when `focus: true` is set (see #8125)
* when examples are filtered by name (`--example` flag)
* when examples are filtered by line (`--line` flag)
* when examples are filtered by location (`--location` flag)
* when examples are filtered by tags (see #8068)

Extracted from #8359

**Comment by asterite** [comment]
When you run `rspec` with `--fail-fast` you only get info of the specs that ran up to the failing test. I don't think we need to show that all of the other specs that didn't run because of `--fail-fast` were skipped.

`skipped` means a test marked as "pending" was part of the run, and that counts toward that. I don't think we should mark as skipped tests not covered by expression filters, lines, etc. So I think this issue is invalid.

**Comment by straight-shoota** [comment]
> `skipped` means a test marked as "pending" was part of the run, and that counts toward that.

No, these are different metrics. Pending tests are counted as `pending`.

`skipped` is supposed to give a count of test that would usually run (and are compiled into the executable), but were excluded for some of the above reasons.

**Comment by asterite** [comment]
Oh... I see, this is about adding a `skipped` counter.

If someone wants to do it, go ahead, but I think it's pretty difficult and it doesn't add much value (at least for me).
